daxa/mission/erosita.py

```diff
@@ -0,0 +1,703 @@
+                            bf=",".join(bad_fields), gf=",".join(self._miss_poss_fields), gft=",".join(self._miss_poss_field_types)))
+        
+        if all(field in self._miss_poss_field_types for field in fields):
+            # Checking if the fields were given as a field type 
+            # Then collecting all the fields associated with that field type/s
+            updated_fields = CalPV_info.loc[CalPV_info["Field_Type"].isin(fields), "Field_Name"].tolist()
+        elif all(field in self._miss_poss_fields for field in fields):
+            # Checking if the field names are valid
+            updated_fields = fields
+        else:
+            # Checking if any of the fields entered are not valid field names or types
+            field_test = [field in self._miss_poss_field_types or self._miss_poss_fields for field in fields]
+            if not all(field_test):
+                bad_fields = np.array(input_fields)[~np.array(field_test)]
+                raise ValueError("Some field names or field types {bf} are not associated with this mission, please"
+                        "choose from the following fields; {gf} or field types; {gft}".format(
+                        bf=",".join(bad_fields), gf=",".join(self._miss_poss_fields), gft=",".join(self._miss_poss_field_types)))
+            
+            else:
+                # If its got to this stage that means the field where input as a mix of field types and field names
+                # I did this when i was tired so sorry it looks a but weird?? i think it might look weird
+                # Selecting the field types from the input
+                field_types = [ft for ft in fields if ft in self._miss_poss_field_types]
+                # Turning them into their field names
+                fields_in_types = CalPV_info.loc[CalPV_info["Field_Type"].isin(field_types), "Field_Name"].tolist()
+                # Selecting the field names from the input
+                field_names = [ft for ft in fields if ft in self._miss_poss_fields]
+                updated_fields = fields_in_types + field_names
+        
+        # Removing the duplicates from updated_fields
+        updated_fields = list(set(updated_fields))
+
+        # Return the chosen fields 
+        return updated_fields
+    
+    def _inst_filtering(self, evlist_path: str):
+        """
+        Method to filter event lists for eROSITACalPV data based on instrument choice.
+        
+        :param Union[List[str], str] gd_tscopes: The names of telescopes from which the user wishes to INCLUDE data from.
+        """
+
+        insts = self.chosen_instruments
+        # Getting a string of the numbers of the telescope modules to add to the file name
+        insts_str = ''.join(sorted(re.findall(r'\d+', ''.join(insts))))
+
+        # Only filtering the eventlist if this combination of instruments hasn't been filtered for yet
+        if os.path.exists(evlist_path[:-5] + '_if_{}.fits'.format(insts_str)):
```

Comment:
I realised I also never updated the docstring param to explain evlist_path either, so I've updated that.

The [-:5] removes the '.fits' part of the file path string, so that it can be replaced with the ending that gets added to the file name after instrument filtering.

Reply:
> I realised I also never updated the docstring param to explain evlist_path either, so I've updated that.
> 
> The [-:5] removes the '.fits' part of the file path string, so that it can be replaced with the ending that gets added to the file name after instrument filtering.

Ah I see of course! Another way of doing that might be evlist_path.split('.')[0] or split('.fits')[0] - not requiring you to change just an FYI